src/main/scala/org/constellation/domain/transaction/TransactionService.scala

```diff
@@ -57,50 +57,77 @@ class TransactionService[F[_]: Concurrent](val transactionChainService: Transact
   def createTransaction(
     src: String,
     dst: String,
+    prevTxRef: String,
+    ordinal: Long,
     amount: Long,
     keyPair: KeyPair,
     normalized: Boolean = true,
     dummy: Boolean = false
-  ): F[Transaction] =
-    TransactionService.createTransaction(src, dst, amount, keyPair, normalized, dummy)(transactionChainService)
+  ): F[Transaction] = TransactionService.createTransaction(src, dst, prevTxRef, ordinal, amount, keyPair, normalized, dummy)(transactionChainService)
 
-  def createDummyTransaction(src: String, dst: String, keyPair: KeyPair): F[Transaction] =
-    TransactionService.createDummyTransaction(src, dst, keyPair)(transactionChainService)
+  def createDummyTransaction(src: String, dst: String, keyPair: KeyPair, prevTxRef: String = "dummy", ordinal: Long = 0L): F[Transaction] =
+    TransactionService.createDummyTransaction(src, dst, keyPair, prevTxRef, ordinal)(transactionChainService)
 
+  def receiveTransaction(tx: Transaction): F[TransactionCacheData] =
+    transactionChainService
+      .setLastTransaction(tx.edge, false)
+      .flatMap(tx => put(TransactionCacheData(tx)))
 }
 
 object TransactionService {
 
   def apply[F[_]: Concurrent](transactionChainService: TransactionChainService[F], dao: DAO) =
     new TransactionService[F](transactionChainService, dao)
 
-  def createTransaction[F[_]: Concurrent](
-    src: String,
-    dst: String,
-    amount: Long,
-    keyPair: KeyPair,
-    normalized: Boolean = true,
-    dummy: Boolean = false
-  )(transactionChainService: TransactionChainService[F]): F[Transaction] = {
+  def createTransactionEdge(
```

Comment:
Please change an option in your IDE to use scalafmt for code formatting

Reply:
Already have it set, looks like I forgot to run Reformat Code